hpx/util/decay.hpp

```diff
@@ -47,7 +28,13 @@ namespace hpx { namespace util
         };
 
         template <typename X>
-        struct decay_unwrap_impl<boost::reference_wrapper<X> >
+        struct decay_unwrap_impl< ::boost::reference_wrapper<X> >
```

Comment:
`::` means global namespace, `::boost::reference_wrapper` is a fully qualified name. It means the same thing as `boost::reference_wrapper` in normal scenarios, but a relative name is sensitive to someone later adding a `boost` entity to any of the intervening namespaces. This is particularly valuable when specializing things in namespaces one does not have control over (which is not quite this case), and I must have carried it from there.

The use of a space between `<` and `::` is intentional just as the use of space between `>` and `>`, as `<:` is a digraph and compilers may not have yet caught up with the C++11 fixes to these parsing oddities.